pyiceberg/io/pyarrow.py

```diff
@@ -1366,6 +1373,7 @@ def project_table(
             case_sensitive,
             table_metadata.name_mapping(),
             use_large_types,
+            limit,
```

Comment:
I think this change is enough to "pushdown the limit". `_task_to_table` will get batches from `_task_to_record_batches` and check the limit, returning a pyarrow Table when the limit is met. 

If the limit is not met, `_task_to_table` will just return all the batches as a pyarrow Table

Reply:
Understood.